packages/components/src/components/tooltip/tooltip.js

```diff
@@ -151,71 +177,183 @@ class Tooltip extends mixin(
     this.tooltip.changeState(
       state,
       Object.assign(detail, { delegatorNode: this.element }),
-      callback
+      () => {
+        // @questions
+        // -- What is the expected behavior if the user clicks on the trigger when the tooltip is already open
+        // -- What's the focus state look like when there is only rich text within a tooltip (no buttons/links/etc)
+        // -- There is a "flash" where the next element (another trigger in the demo) is focused before focus is
+        //      moved to the tooltip. I'm assuming this is due to the debounce. Now that it's triggered on "click"
+        //      for keyboard users do we still want the debounce?
+        this.isHidden = state !== 'shown';
```

Comment:
Instead of caching hidden state (that tends to create multiple sources of truth), we tend to query for whether a class is applied to an element, etc. to query for state.